orderbook-rs/src/interpreter/mod.rs

```diff
@@ -0,0 +1,112 @@
+use crate::registry::{IExpressionDeployerV3, IParserV1};
+use alloy_primitives::{Address, Bytes, U256};
+use anyhow::anyhow;
+use ethers::{
+    providers::{Http, Provider},
+    types::H160,
+};
+use std::str::FromStr;
+use std::sync::Arc;
+use tracing::error;
+
+/// Get RainInterpreterNPE2, RainterpreterStoreNPE2 and RainterpreterParserNPE2 addresses corresponding to a RainterpreterExpressionDeployerNPE2 contract.
+///
+/// # Arguments
+/// * `deployer_npe2` - Address of RainterpreterExpressionDeployerNPE2.
+/// * `rpc_url` - Network RPC URL.
+///
+pub async fn get_disp(
+    deployer_npe2: Address,
+    rpc_url: String,
+) -> anyhow::Result<(Address, Address, Address)> {
+    let provider = match Provider::<Http>::try_from(rpc_url.clone()) {
+        Ok(provider) => provider,
+        Err(err) => {
+            error!("INVALID RPC URL");
```

Comment:
probably if this is to be a crate we want an error enum rather than strings, generally anyhow is probably better for binaries than libs

this is because if we expect someone else to use this crate as a dependency, they may want to change the behaviour of their code based on what happens in this code, in which case an enum allows them to do something like `match error { RainOrderbookError::InvalidRPC => { ... } }`

Reply:
Added error enum.